src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -1,15 +1,127 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import isValidEntityId from "../../../common/entity/valid_entity_id.js";
+import { fireEvent } from "../../../common/dom/fire_event.js";
 
 import "../../../components/ha-card.js";
 
-import EventsMixin from "../../../mixins/events-mixin.js";
+interface Config extends LovelaceConfig {
+  entity: string;
+  title?: string;
+  unit_of_measurement?: string;
+  min?: number;
+  max?: number;
+  severity?: object;
+}
+
+class HuiGaugeCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error("Invalid card configuration");
+    }
+    if (!isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+    this._config = { min: 0, max: 100, ...config };
+  }
+
+  protected render() {
+    if (!this._config) {
```

Comment:
You should guard for hass being undefined too, otherwise line 47 will fail. When you add `!` to a variable name to silent TypeScript, you better be damn sure that the variable always exists.